homeassistant/components/hunterdouglas_powerview/config_flow.py

```diff
@@ -113,6 +113,7 @@ async def async_step_discovery_confirm(self):
         """Confirm dhcp or homekit discovery."""
         # If we already have the host configured do
         # not open connections to it if we can avoid it.
+        self.context[CONF_HOST] = self.discovered_ip
         for progress in self._async_in_progress():
             if progress.get("context", {}).get(CONF_HOST) == self.discovered_ip:
```

Comment:
Do you need the .get when you set CONF_HOST before the loop?
```suggestion
            if progress.get("context", {})[CONF_HOST] == self.discovered_ip:
```

Reply:
Yes because there could be a non discovery flow in progress

Reply:
you are right